GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Branches.cs

```diff
@@ -281,8 +281,11 @@ private Nodes FillBranchTree(IEnumerable<string> branches, CancellationToken tok
                 #endregion
 
                 var nodes = new Nodes(this);
+
+                token.ThrowIfCancellationRequested();
                 var aheadBehindData = _aheadBehindDataProvider?.GetData();
```

Comment:
❔ Why not just pass `token` to `GetData()`?

Reply:
`FillBranchTree` is an integration function. It checks the token before every call to git now.
`GetData` issues a single git command. Why bloat the interface? (Separation of concerns.)

Reply:
This approach fails to correctly handle cancellation in the middle of the long-running Git operation. The interface is incorrect and eventually needs to be updated. Until then, checking the cancellation token only before the long-running operation starts does not offer much value.

Reply:
OK, I have opened the new issue #7447 for this requested extension.
I added the calls just for consistency, which is better than without before. It does not affect the stability of the test runs, i.e. it is not really related to this PR.